src/Components/Analyzers/src/ComponentParametersShouldBePublicCodeFixProvider.cs

```diff
@@ -64,9 +64,24 @@ private SyntaxNode HandlePropertyDeclaration(PropertyDeclarationSyntax node)
                 return null;
             }
 
-            var publicModifier = node.Modifiers.FirstOrDefault(m => m.IsKind(SyntaxKind.PublicKeyword));
-            node = node.WithModifiers(
-                node.Modifiers.Remove(publicModifier));
+            var newModifiers = node.Modifiers;
+            for (var i = 0; i < node.Modifiers.Count; i++)
+            {
+                var modifier = node.Modifiers[i];
+                if (modifier.IsKind(SyntaxKind.PrivateKeyword) ||
+                    modifier.IsKind(SyntaxKind.ProtectedKeyword) ||
+                    modifier.IsKind(SyntaxKind.InternalKeyword) ||
+
+                    // We also remove public in case the user has written something totally backwards such as private public protected Foo
+                    modifier.IsKind(SyntaxKind.PublicKeyword))
+                {
+                    newModifiers = newModifiers.Remove(modifier);
+                }
+            }
```

Comment:
Should we remove modifiers on the getter and setter?

```C#
[Parameter] string Foo { get; private set;  }
```

This will still result in an error after running the code fix. That would be annoying.

Reply:
> * Changed the code fix provider to no longer worry about the property setters and instead inspect the properties declared accessibility. Did not re-add the property setter code fix provider because these code fix providers do nothing in practice (Razor light bulbs aren't enabled in the editor).